app/routes/route_utilities.py

```diff
@@ -0,0 +1,31 @@
+from flask import abort, make_response
+from ..db import db 
+
+def validate_model(cls, id):
+    try:
+        id = int(id)
+    except ValueError: 
+        response = {"details": f"{cls.__name__} {id} invalid"}
+        abort(make_response(response, 400))
+    
+    query = db.select(cls).where(cls.id == id)
+    model = db.session.scalar(query)
+
+    if not model:
+        response = {"details": f"{cls.__name__} {id} not found"}
+        abort(make_response(response, 404))
+    
+    return model
+
+
+def create_model(cls, model_data):
+    try:
+        new_model = cls.from_dict(model_data)
+    except KeyError as e:
+        response = {"details": "Invalid data"}
```

Comment:
It might be nice to provide even more details in the response you send back so that the client knows how to fix up the request body they will re-send.

```suggestion
        response = {"details":  f"Invalid data for field: {error.args[0]}"}
```

For example, this would evaluate to "title" for example if I sent a bad request body to create a Task.